diff --git a/dist/NodeFileSystem.js b/dist/NodeFileSystem.js
index f2230ae56a9161e2d45d3cf9f2c0a79d8d08a61c..d347332d8e5fa852d8127fe4707212e37a79f792 100644
--- a/dist/NodeFileSystem.js
+++ b/dist/NodeFileSystem.js
@@ -335,9 +335,9 @@ const utimes = /*#__PURE__*/(() => {
   return (path, atime, mtime) => nodeUtimes(path, atime, mtime);
 })();
 // == watch
-const watchNode = path => Stream.callback(queue => Effect.acquireRelease(Effect.sync(() => {
+const watchNode = (path, options) => Stream.callback(queue => Effect.acquireRelease(Effect.sync(() => {
   const watcher = NFS.watch(path, {
-    recursive: true
+    recursive: options?.recursive ?? false
   }, (event, path) => {
     if (!path) return;
     switch (event) {
@@ -379,7 +379,7 @@ const watchNode = path => Stream.callback(queue => Effect.acquireRelease(Effect.
   });
   return watcher;
 }), watcher => Effect.sync(() => watcher.close())));
-const watch = (backend, path) => stat(path).pipe(Effect.map(stat => backend.pipe(Option.flatMap(_ => _.register(path, stat)), Option.getOrElse(() => watchNode(path)))), Stream.unwrap);
+const watch = (backend, path, options) => stat(path).pipe(Effect.map(stat => backend.pipe(Option.flatMap(_ => _.register(path, stat, options)), Option.getOrElse(() => watchNode(path, options)))), Stream.unwrap);
 // == writeFile
 const writeFile = (path, data, options) => Effect.callback((resume, signal) => {
   try {
@@ -422,8 +422,8 @@ const makeFileSystem = /*#__PURE__*/Effect.map(/*#__PURE__*/Effect.serviceOption
   symlink,
   truncate,
   utimes,
-  watch(path) {
-    return watch(backend, path);
+  watch(path, options) {
+    return watch(backend, path, options);
   },
   writeFile
 }));
diff --git a/src/NodeFileSystem.ts b/src/NodeFileSystem.ts
index fe7dd6314935ac8969f6e3b8ae4c2c6d3b229655..c9a7ecb4fa1a791faf43cd4ed2c1a13bf057b65c 100644
--- a/src/NodeFileSystem.ts
+++ b/src/NodeFileSystem.ts
@@ -550,12 +550,12 @@ const utimes = (() => {
 
 // == watch
 
-const watchNode = (path: string) =>
+const watchNode = (path: string, options?: FileSystem.WatchOptions) =>
   Stream.callback<FileSystem.WatchEvent, Error.PlatformError>((queue) =>
     Effect.acquireRelease(
       Effect.sync(() => {
         const watcher = NFS.watch(path, {
-          recursive: true
+          recursive: options?.recursive ?? false
         }, (event, path) => {
           if (!path) return
           switch (event) {
@@ -595,12 +595,16 @@ const watchNode = (path: string) =>
     )
   )
 
-const watch = (backend: Option.Option<FileSystem.WatchBackend["Service"]>, path: string) =>
+const watch = (
+  backend: Option.Option<FileSystem.WatchBackend["Service"]>,
+  path: string,
+  options?: FileSystem.WatchOptions
+) =>
   stat(path).pipe(
     Effect.map((stat) =>
       backend.pipe(
-        Option.flatMap((_) => _.register(path, stat)),
-        Option.getOrElse(() => watchNode(path))
+        Option.flatMap((_) => _.register(path, stat, options)),
+        Option.getOrElse(() => watchNode(path, options))
       )
     ),
     Stream.unwrap
@@ -652,8 +656,8 @@ const makeFileSystem = Effect.map(Effect.serviceOption(FileSystem.WatchBackend),
     symlink,
     truncate,
     utimes,
-    watch(path) {
-      return watch(backend, path)
+    watch(path, options) {
+      return watch(backend, path, options)
     },
     writeFile
   }))
